app/src/org/commcare/services/CommCareSessionService.java

```diff
@@ -301,6 +303,11 @@ public void run() {
      * progess then don't do anything.
      */
     private void timeToExpireSession() {
+        if (CommCareApplication._().isConsumerApp()) {
+            // Never expire the session if this is a consumer app
+            return;
+        }

```

Comment:
Instead of doing this can you just not setup/run `maintenanceTimer` if it is a consumer app?